### PR TITLE
Basic vsnd_c phoneme reading

### PR DESCRIPTION
**Pull request by kristiker**
Related to this https://developer.valvesoftware.com/wiki/Phoneme_Tool/data_format

Only reads the WORDS section, I couldn't figure out how other stuff is packed.

### CODE REVIEW

File: ValveResourceFormat/Resource/ResourceTypes/Sound.cs [hunk 1]
```diff
@@ -144,13 +169,19 @@ public override void Read(BinaryReader reader, Resource resource)
             SampleCount = reader.ReadUInt32();
             Duration = reader.ReadSingle();
 
-            // Skipping over m_Sentence (CSentence_t) and m_pHeader
-            reader.BaseStream.Position += 12;
+            var sentenceOffset = reader.ReadInt64();
```
Comment (xPaw) [inline ValveResourceFormat/Resource/ResourceTypes/Sound.cs:172]: Is it actually long?

File: ValveResourceFormat/Resource/ResourceTypes/Sound.cs [hunk 2]
```diff
@@ -144,13 +169,19 @@ public override void Read(BinaryReader reader, Resource resource)
...
+            
+            if (sentenceOffset != 0)
```
Comment (xPaw) [inline ValveResourceFormat/Resource/ResourceTypes/Sound.cs:174]: Add brackets, and run format document in VS

File: ValveResourceFormat/Resource/ResourceTypes/Sound.cs [hunk 3]
```diff
@@ -144,13 +169,19 @@ public override void Read(BinaryReader reader, Resource resource)
...
+                sentenceOffset = reader.BaseStream.Position + sentenceOffset;
+            
+            // Skipping over m_pHeader
+            reader.BaseStream.Position += 4;
 
             StreamingDataSize = reader.ReadUInt32();
 
             if (resource.Version < 1)
```
Comment (xPaw) [inline ValveResourceFormat/Resource/ResourceTypes/Sound.cs:182]: Can you flip the ifs so there's no early returns?

File: ValveResourceFormat/Resource/ResourceTypes/Sound.cs [hunk 4]
```diff
@@ -179,12 +211,51 @@ public override void Read(BinaryReader reader, Resource resource)
...
-            LoopEnd = reader.ReadInt32();
+            Sentence = new Sentence();
+
+            Sentence.RunTimePhonemes = new PhonemeTag[numPhonemeTags];
+
+            for (var i = 0; i < numPhonemeTags; i++)
+            {
+                var startTime = reader.ReadSingle();
+                var endTime = reader.ReadSingle();
+                var phonemeCode = (UInt16)reader.ReadInt32();
```
Comment (xPaw) [inline ValveResourceFormat/Resource/ResourceTypes/Sound.cs:248]: Why?

Comment (kristiker) [reply ValveResourceFormat/Resource/ResourceTypes/Sound.cs:248]: Valve aligns it to 4 bytes. Should I really ReadUInt16 and skip 2?

File: ValveResourceFormat/Resource/ResourceTypes/Sound.cs (hunk 1 above)
Comment (kristiker) [reply ValveResourceFormat/Resource/ResourceTypes/Sound.cs:172]: Yeah probably isn't. Will revert it now. Reads fine though on all vsnds (next int is always 0x00).

File: ValveResourceFormat/Resource/ResourceTypes/Sound.cs (hunk 4 above)
Comment (xPaw) [reply ValveResourceFormat/Resource/ResourceTypes/Sound.cs:251]: If it's alignment bytes, skip them separately.

File: ValveResourceFormat/Resource/ResourceTypes/Sound.cs (hunk 1 above)
Comment (xPaw) [reply ValveResourceFormat/Resource/ResourceTypes/Sound.cs:172]: There are a bunch of zeroes, like all the others where we check for zero and throw if it's not (to find any).

File: ValveResourceFormat/Resource/ResourceTypes/Sound.cs (hunk 3 above)
Comment (kristiker) [reply ValveResourceFormat/Resource/ResourceTypes/Sound.cs]: Done.

**Pull request merged**